src/middlewared/middlewared/plugins/failover_/event.py

```diff
@@ -511,6 +511,16 @@ def vrrp_master(self, job, fobj, ifname, event):
         logger.info('Configuring system dataset')
         self.run_call('systemdataset.setup')
 
+        smb_setup = self.run_call(
+            'core.get_jobs',
+            [('method', '=', 'smb.configure')],
+            {'order_by': ['-id']}
+        )
+
+        if not smb_setup:
```

Comment:
Am I going crazy? Should this not be `if smb_setup:` ?

Also, can you add a brief comment on why we do this since `systemdataset.setup` returns before it's actually complete...

Reply:
Adding the comment is for my feeble brain so that I know why we're doing this on failover 😄

Reply:
No, you're not going crazy, I am.